frontend/src/components/compiler/CompilerOpts.tsx

```diff
@@ -55,58 +58,41 @@ export function FlagOption({ flag, description }: { flag: string, description?:
     </option>
 }
 
-export type CompilerOptsT = {
-    compiler: string
-    compiler_flags: string
-}
-
 export type Props = {
+    compiler: string
+    flags: string
+    onCompilerChange: (compiler: string) => void
+    onFlagsChange: (flags: string) => void
     platform?: string
-    value: CompilerOptsT
-    onChange: (value: CompilerOptsT) => void
     title?: string
     isPopup?: boolean
 }
 
-export default function CompilerOpts({ platform, value, onChange, title, isPopup }: Props) {
-    const compiler = value.compiler
-    let opts = value.compiler_flags
-
-    const setCompiler = (compiler: string) => {
-        onChange({
-            compiler,
-            compiler_flags: opts,
-        })
-    }
-
-    const setOpts = (opts: string) => {
-        onChange({
-            compiler,
-            compiler_flags: opts,
-        })
-    }
-
+export default function CompilerOpts({ compiler, flags, onCompilerChange, onFlagsChange, platform, title, isPopup }: Props) {
     return <OptsContext.Provider value={{
         checkFlag(flag: string) {
-            return (" " + opts + " ").includes(" " + flag + " ")
+            return flags.split(" ").includes(flag)
```

Comment:
Doesn't this break with args that are multi part like `- foo 123`? This is why I did the whitespace approach

Reply:
Ah, yes, good point